android/app/src/main/java/org/fossasia/openevent/viewmodels/auth/SignUpActivityViewModel.java

```diff
@@ -0,0 +1,60 @@
+package org.fossasia.openevent.viewmodels.auth;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.utils.AuthUtil;
+import org.fossasia.openevent.utils.Utils;
+
+public class SignUpActivityViewModel extends ViewModel {
+    public static final int VALID = 0;
+    public static final int EMPTY = 1;
+    public static final int INVALID = 2;
+
+    private MutableLiveData<Integer> signUpResponse;
+    private MutableLiveData<Integer> loginResponse;
+
+    public int validateEmail(String email) {
+       if (Utils.isEmpty(email)) {
+           return EMPTY;
+       } else if (!Utils.isEmailValid(email)) {
+           return INVALID;
+       }
+       return VALID;
+    }
+
+    public int validatePassword(String password) {
+        if (Utils.isEmpty(password)) {
+            return EMPTY;
+        } else if (!Utils.isPasswordValid(password)) {
+            return INVALID;
+        }
+        return VALID;
+    }
+
+    public int validateConfirmPassword(String confirmPassword, String password) {
+        if (Utils.isEmpty(confirmPassword)) {
+            return EMPTY;
+        } else if (!confirmPassword.equals(password)) {
+            return INVALID;
+        }
+        return VALID;
+    }
+
+    public LiveData<Integer> signUpUser(String email, String password) {
+        if (signUpResponse == null) {
+            signUpResponse = new MutableLiveData<>();
+        }
+        AuthUtil.signUpUser(email, password, signUpResponse);
+        return signUpResponse;
+    }
+
+    public LiveData<Integer> loginUserAfterSignUp(String email, String password) {
+        if (loginResponse == null) {
+            loginResponse = new MutableLiveData<>();
+        }
+        AuthUtil.loginUserWithoutProgressBar(email, password, loginResponse);
```

Comment:
If this is not used anywhere else. Move this in here please

Reply:
It will be used in the future by LoginActivity. 
I had to refactor both the signUp and loginUser methods. signUp user method was only used by SignUpActivity so i did it. The loginUser method was also used by the LoginActivity, hence i couldn't afford editing that. Hence i created a temporary method loginUserWithoutProgressBar to serve the purpose.
The idea is to remove the loginUser method and rename this loginUserWithoutProgressBar method to loginUser and make the LoginActivity use this method using it's ViewModel once it is created.